### DISCUSSION TITLE
Add authentication in the proxy

### DISCUSSION THREAD

**Discussion by liamskydamien** [Unleash/unleash#7461] [Q&A] [answered]
Hello,
I'm currently investigating into Unleash and Unleash Proxy. 
Can I add an authentication procedure in the unleash proxy? I need to verify a JWT and attach a custom context based on that. I cannot add the context on the client side, it needs to be done on the server side.

Is something like this possible?

**Comment by chriswk** [comment, marked as answer]
Hi, the proxy app allows you to add a preHook to your config when you start it up. So something like
```javascript
const jwt = require('jsonwebtoken');
const proxy = require('unleash-proxy');
const getCustomData = require('./myContextBuilder'); // This you have to write yourself.
const config = proxy.createProxyConfig({
 preHook: (app) => {
   app.use('/', (req, res, next) => {
 // extract token from req
// validate
// populate req query with extra parameters to use in context enricher
req.query.jwtValidated = true
});
},
 expContextEnricher: [(context) => {
     if (context.parameters.jwtValidated === true) { 
        context.myCustomContext = getCustomData() 
     }}]
});

const myApp = proxy.createApp(config);
myApp.start();
```
Should work. Though the above example probably won't build it should give you pointers to what you'll need to add.

**Reply by liamskydamien** [reply]
Thank you very much!
I will try it out😄
